core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/ValidationCleanupService.java

```diff
@@ -0,0 +1,75 @@
+package com.b2international.snowowl.core.branch;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.core.runtime.jobs.ISchedulingRule;
+
+import com.b2international.snowowl.core.ServiceProvider;
+import com.b2international.snowowl.core.events.AsyncRequest;
+import com.b2international.snowowl.core.identity.User;
+import com.b2international.snowowl.core.jobs.JobRequests;
+import com.b2international.snowowl.core.jobs.SerializableSchedulingRule;
+import com.b2international.snowowl.core.validation.ValidationRequests;
+import com.google.common.base.Joiner;
+
+public class ValidationCleanupService {
+	
+	public static final SerializableSchedulingRule VALIDATION_ISSUE_CLEANUP_RULE = new SerializableSchedulingRule() {
+		
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		public boolean isConflicting(ISchedulingRule rule) {
+			return rule == this;
+		}
+		
+		@Override
+		public boolean contains(ISchedulingRule rule) {
+			return rule == this;
+		}
+	};
+
+	public ValidationCleanupService() {
+	}
+	
+	public void removeStaleIssues(ServiceProvider context, String resourceURI) {
+		removeStaleIssues(context, List.of(resourceURI), Collections.emptyList());
+	}
+	
+	public void removeStaleIssues(ServiceProvider context, List<String> resourceURIs) {
+		removeStaleIssues(context, resourceURIs, Collections.emptyList());
+	}
+	
+	public void removeStaleIssues(ServiceProvider context, String resourceURI, List<String> previousAnalysisRunIds) {
+		removeStaleIssues(context, List.of(resourceURI), previousAnalysisRunIds);
+	}
+	
+	public void removeStaleIssues(ServiceProvider context, List<String> resourceURIs, List<String> previousAnalysisRunIds) {
+		AsyncRequest<Boolean> request;
+		
+		if (!previousAnalysisRunIds.isEmpty()) {
+			request = ValidationRequests.issues()
+					.prepareDelete()
+					.setResultIds(previousAnalysisRunIds)
+					.buildAsync();
+		} else {
+			request = ValidationRequests.issues()
+				.prepareDelete()
+				.setCodeSystemURIs(resourceURIs)
+				.buildAsync();
+		}
+		
+		final String description = String.format("Remove validation issues on stale/removed branch(es) %s", Joiner.on(", ").join(resourceURIs));
+		
+		JobRequests.prepareSchedule()
+			.setRequest(request)
+			.setDescription(description)
+			.setUser(context.service(User.class).getUserId())
```

Comment:
This is the default behavior in `ScheduleJobRequest`, you can leave this property unset if you don't want to override the requester: https://github.com/b2ihealthcare/snow-owl/blob/6b2f0c92b5dbe9563a337ae4869a795cd10f2b01/core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/jobs/ScheduleJobRequest.java#L125